include/simdjson/generic/numberparsing.h

```diff
@@ -582,7 +583,7 @@ simdjson_inline error_code parse_number(const uint8_t *const src, W &writer) {
   }
   if (is_float) {
     const bool dirty_end = jsoncharutils::is_not_structural_or_whitespace(*p);
```

Comment:
1. This PR's commit title and documentation could be clarified - if the intent is to add support for arbitrary-length *floating point* numbers, it should say that instead of arbitrary-length numbers

    Right now, it'll continue to reject integers out of the signed+unsigned 64-bit range, and floating point numbers outside of the largest possible finite `double` min/max range?
2. If the intent is to also support parsing arbitrary-length integers (e.g. `"100000000000000000000000000000000000000000"`), maybe there could be a build flag or parser instance property option to support `goto is_float_handler` instead of `return INVALID_NUMBER(src);` for the following checks (for example, if this were to be used in bindings supporting something similar to PHP's JSON_BIGINT_AS_STRING)

   1. `if (digit_count > longest_digit_count) { return INVALID_NUMBER(src); }`
   2. `}  else if (src[0] != uint8_t('1') || i <= uint64_t(INT64_MAX)) { return INVALID_NUMBER(src); }`